### PR TITLE
feat: add 'event' tracking type to support simplified event tracking

### PR DESCRIPTION
**Pull request by JasonBerry**

### CODE REVIEW

File: src/defaults.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ interface MergePropsParams extends SearchResultsProps {
 
 interface MergedSearchResultsProps extends Omit<SearchResultsProps, 'options' | 'tracking' | 'preset'> {
   options: SearchResultsOptions;
-  tracking?: ClickTracking | PosNegTracking;
+  tracking?: ClickTracking | PosNegTracking | EventTracking;
```
Comment (wwalser) [inline src/defaults.ts:19]: Can the type be `Tracking` here or is that insufficient?

File: src/types/index.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ export interface SearchWidgetBaseOptions {
   pipeline: PipelineOption;
   endpoint?: string;
   clickTokenURL?: string;
+  searchIOAnalyticsEndpoint?: string;
```
Comment (wwalser) [inline src/types/index.ts:44]: This could be a part of the tracking options object L47?

**Review by wwalser — APPROVED** [review]

File: src/defaults.ts (hunk 1 above)
Comment (JasonBerry) [reply src/defaults.ts:19]: That would make `NoTracking` also valid, so I wanted to keep it more restricted.

File: src/types/index.ts (hunk 2 above)
Comment (JasonBerry) [reply src/types/index.ts:44]: I was keeping it consistent with where `clickTokenURL` is configured. I don't have strong feelings about it either way.

Comment (wwalser) [reply src/types/index.ts:44]: Nah, just noting.

**Comment by wwalser** [comment]
Fire away. Builds should run in release PRs now as well. 🤞

**Pull request merged**